internal/worker/json.go

```diff
@@ -72,6 +73,19 @@ type KojiFinalizeJobResult struct {
 	KojiError string `json:"koji_error"`
 }
 
+type DepsolveJob struct {
+	PackageSets      map[string]rpmmd.PackageSet `json:"package_sets"`
+	Repos            []rpmmd.RepoConfig          `json:"repos"`
+	ModulePlatformID string                      `json:"module_platform_id"`
+	Arch             string                      `json:"arch"`
+	Releasever       string                      `json:"releasever"`
+}
+
+type DepsolveJobResult struct {
+	PackageSpecs map[string][]rpmmd.PackageSpec `json:"package_specs"`
```

Comment:
I think using structs from another package for serialisation is an anti-pattern we should try to avoid (who even wrote this patch?), but I would not consider this a blocker, as this would anyway need more cleanup across the codebase.